src/nspawn/nspawn.c

```diff
@@ -2218,6 +2285,29 @@ static int setup_machine_id(const char *directory) {
         return 0;
 }
 
+static int recursive_chown(const char *directory, uid_t shift, uid_t range) {
+        int r;
+
+        assert(directory);
+
+        if (arg_userns_mode == USER_NAMESPACE_NO || !arg_userns_chown)
+                return 0;
+
+        r = path_patch_uid(directory, arg_uid_shift, arg_uid_range);
+        if (r == -EOPNOTSUPP)
+                return log_error_errno(r, "Automatic UID/GID adjusting is only supported for UID/GID ranges starting at multiples of 2^16 with a range of 2^16.");
+        if (r == -EBADE)
+                return log_error_errno(r, "Upper 16 bits of root directory UID and GID do not match.");
+        if (r < 0)
+                return log_error_errno(r, "Failed to adjust UID/GID shift of OS tree: %m");
```

Comment:
Having another look: I really don't grok your strace output... the syscall_r326 appears to be done in a different process? seems unrelated to the openat?

That said, if it is indeed /sys that is a read-only tmpfs for you, then I figure the problem is that the files in it cannot be chowned. I don't understand why you mount /sys like this though. Why not leave that to nspawn?

If you premount it, you should probably make sure the dir is owned by the future root in the container... which of course is hard if it is randomly assigned by nspawn.